core/generator.ts

```diff
@@ -441,6 +437,10 @@ export class CodeGenerator {
     return msg.replace(/%1/g, "'" + id + "'");
   }
 
+  abstract quote_(toQuote: string): string;
+
+  abstract multiline_quote_(toQuote: string): string;
```

Comment:
Admittedly this one is harder to provide a generic implementation for.  It could just default to using `quote` (and we could fix `quote` to handle newlines correctly), losing the nice formatting.